### PR TITLE
Cjavilla/unify pmd

### PR DESCRIPTION
**Pull request by cjavilla-stripe**
- [x] ⚠️ Stop sharing from `PaymentIntentPaymentMethodDataOptions`
- [x] ⚠️ Stop sharing sub hashes from `PaymentMethodCreateOptions` and `PaymentMethodUpdateOptions`

I did this in two commits to try to make the review a little easier. I think this one likely needs a bit bigger blurb in the change log.

r? @remi-stripe 
cc @stripe/api-libraries

### CODE REVIEW

File: src/Stripe.net/Services/PaymentMethods/PaymentMethodBancontactOptions.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+namespace Stripe
```
Comment (remi-stripe) [inline src/Stripe.net/Services/PaymentMethods/PaymentMethodBancontactOptions.cs:1]: it's weird it doesn't see this one as a rename

File: src/Stripe.net/Services/PaymentMethods/PaymentMethodCardOptions.cs [hunk 2]
```diff
@@ -2,7 +2,7 @@ namespace Stripe
 {
     using Newtonsoft.Json;
 
-    public class PaymentMethodCardCreateOptions : INestedOptions
+    public class PaymentMethodCardOptions : INestedOptions
```
Comment (remi-stripe) [inline src/Stripe.net/Services/PaymentMethods/PaymentMethodCardOptions.cs:5]: For later it'd be need if we flagged params that are only on create (or vice versa) in the docstring. Not worth it but it'd be a good todo to track

Comment (cjavilla-stripe) [reply src/Stripe.net/Services/PaymentMethods/PaymentMethodCardOptions.cs:5]: yeah I think we could likely do that in the params somehow. Right now, I'm merging the params between create and update pretty far up-stream, so we'd likely need to pass down the state from pretty early on.

File: src/Stripe.net/Services/PaymentMethods/PaymentMethodBancontactOptions.cs (hunk 1 above)
Comment (cjavilla-stripe) [reply src/Stripe.net/Services/PaymentMethods/PaymentMethodBancontactOptions.cs:1]: yeah I think between the class name changing, and removing the unneeded json using statement, it's different enough to git to think it's a new file 🤷

File: src/Stripe.net/Services/PaymentIntents/PaymentIntentPaymentMethodDataAuBecsDebitOptions.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+namespace Stripe
+{
+    using Newtonsoft.Json;
+
+    public class PaymentIntentPaymentMethodDataAuBecsDebitOptions : INestedOptions
+    {
+        [JsonProperty("account_number")]
```
[the remark this answers is not in the document]
Comment (cjavilla-stripe) [reply src/Stripe.net/Services/PaymentIntents/PaymentIntentPaymentMethodDataAuBecsDebitOptions.cs:10]: Added :)

**Pull request merged**